File: C++_Python_OpenCV/OpenCV_01_Loading.cpp
// Including the 2 usefull libraries
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
	
    // Print OpenCV version
    std::cout << "OpenCV Version: " << CV_VERSION << std::endl;
    
    // Reading the demo image and saving it in the Mat 'img'
	cv::Mat img = cv::imread("demo_image.png", cv::IMREAD_COLOR);
	
	// Reading the demo image in grayscale and saving it
	cv::Mat img_gray = cv::imread("demo_image.png", cv::IMREAD_GRAYSCALE);

    // Alternatively:
	// IMREAD_COLOR: 1
	// IMREAD_GRAYSCALE: 0
	// IMREAD_UNCHANGED: -1

	// Create a window.
	cv::namedWindow( "Demo image", cv::WINDOW_AUTOSIZE );
	cv::namedWindow( "Grayscale image", cv::WINDOW_AUTOSIZE );

	// Show the image inside it.
	cv::imshow( "Demo image", img ); 
	cv::imshow( "Grayscale image", img_gray ); 
 
	// Wait for a keystroke.   
	cv::waitKey(0);  
 
	// Destroys all the windows created                         
	cv::destroyAllWindows();
	
    return 0;
}

File: C++_Python_OpenCV/OpenCV_01_Loading.py
# Importing the cv2 library as cv
import cv2

# Printing the version of OpenCV
print("OpenCV version:", cv2.__version__)

# Reading the demo image and saving it in the variable 'img'
img = cv2.imread('demo_image.png',cv2.IMREAD_COLOR)

# Reading the demo image in grayscale and saving it
img_gray = cv2.imread('demo_image.png',cv2.IMREAD_GRAYSCALE)

# Alternatively:
# IMREAD_COLOR: 1
# IMREAD_GRAYSCALE: 0
# IMREAD_UNCHANGED: -1

# Displaying the images inside a window
cv2.imshow('Demo image',img)  
cv2.imshow('Grayscale image',img_gray)
 
# Waiting for a keystroke
cv2.waitKey(0)  
 
# Destroying all the windows created
cv2.destroyAllWindows()
